src/gallia/udscan/scanner/fuzz_payloads.py

```diff
@@ -0,0 +1,192 @@
+import binascii
+import random
+from argparse import Namespace
+
+from gallia.transports.base import TargetURI
+from gallia.transports.can import RawCANTransport
+from gallia.uds.core.client import UDSErrorCodes, UDSRequestConfig
+from gallia.uds.core.exception import IllegalResponse
+from gallia.uds.core.service import NegativeResponse, UDSResponse
+from gallia.uds.helpers import suggests_identifier_not_supported
+from gallia.udscan.core import UDSScanner
+from gallia.utils import auto_int
+
+
+class FuzzPayloads(UDSScanner):
+    def __init__(self) -> None:
+        super().__init__()
+        self.can_recv_event = asyncio.Event()
+
+    def add_parser(self) -> None:
+        self.parser.add_argument(
+            "--sessions",
+            type=auto_int,
+            default=[1],
+            nargs="*",
+            help="Set list of sessions to be tested; 0x01 if None",
+        )
+        self.parser.add_argument(
+            "--serviceid",
+            type=auto_int,
+            default=0x2E,
+            choices=[0x2E, 0x31],
+            help="""
+            Service ID to create payload for; defaults to 0x2e WriteDataByIdentifier;
+            currently supported:
+            0x2e WriteDataByIdentifier, 0x31 RoutineControl
+            """,
+        )
+        self.parser.add_argument(
+            "--max-length",
+            type=auto_int,
+            default=42,
+            help="maximum length of the payload",
+        )
+        self.parser.add_argument(
+            "--min-length",
+            type=auto_int,
+            default=1,
+            help="minimum length of the payload",
+        )
+        self.parser.add_argument(
+            "-i",
+            "--iterations",
+            type=auto_int,
+            default=1,
+            help="number of iterations",
+        )
+        self.parser.add_argument(
+            "--dids",
+            type=auto_int,
+            nargs="*",
+            required=True,
+            help="data identifiers to fuzz",
+        )
+        self.parser.add_argument(
+            "--prefixed-payload",
+            metavar="HEXSTRING",
+            type=binascii.unhexlify,
+            default=b"",
+            help="static payload, which precedes the fuzzed payload",
+        )
+        self.parser.add_argument(
+            "--observe-can-ids",
+            type=auto_int,
+            default=[],
+            nargs="*",
+            help="can ids to observe while fuzzing",
+        )
+
+    def generate_payload(self, args: Namespace) -> bytes:
+        return random.randbytes(random.randint(args.min_length, args.max_length))
+
+    async def observe_can_messages(self, can_ids: list[int], args: Namespace) -> None:
+
+        can_url = args.target.url._replace(scheme=RawCANTransport.SCHEME)
+        transport = RawCANTransport(TargetURI(can_url.geturl()))
+        await transport.connect()
+        transport.set_filter(can_ids, inv_filter=False)
+
+        self.can_recv_event.set()
```

Comment:
The event is never read in the main loop, isn't it?

Reply:
No, it doesn't. The event is not read. The log message is generated in a background task which is created in line 123. I suggest removing the event.